keps/sig-node/4191-split-image-filesystem/README.md

```diff
@@ -0,0 +1,923 @@
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+### Version Skew Strategy
+
+The initial release of this will be the CRI API and changes to Kubelet.  
+
+We do not assume that container runtimes must implement this API so we will assume a single filesystem for images.
+
+Once the container runtimes implement this API, then the feature would be active.
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and nodes?
+- How does an n-3 kubelet or kube-proxy without this feature available behave when this feature is used?
+- How does an n-1 kube-controller-manager or kube-scheduler without this feature available behave when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+If a container runtime is configured to store these separately, there is no really good way to roll these changes back.
+We will include a feature gate for best practices to guard against our code.
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: KubeletSeparateDiskGC
+  - Components depending on the feature gate: Kubelet
+- [x] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane? Yes.
```

Comment:
This is actually only true if the control plane happens to run on a kubelet, right?
It requires downtime of the dataplane (kubelets) but not strictly speaking API server/etcd ... even though taking down all kubelest may cause control plane downtime in some clusters.

Whereas below is also the opposite? Enabling this requires node downtime because kubelet must be restarted?

Reply:
Updated. PTAL